Convert all operations to prefect flows

dgarros opened the pull request:
To help with the transition to prefect,
This PR adds the decorator `@flow` to all existing asynchronous tasks / operations. The main benefit is that all executions will be tracked in prefect which will significantly improve the visibility of all tasks.

dgarros committed:
Convert all operations to prefect flow

dgarros committed:
Move InfrahubServices out of TYPE_CHECKING block

dgarros committed:
Skip prefect when using BusSimulator

dgarros committed:
remove tasks and adjust unit tests and http mock is in use

ogenstad reviewed (approved):
What do you think of adding something like this? Looks like some tasks doesn't have the decorator but I don't know if this is on purpose.

```diff
diff --git a/backend/tests/unit/message_bus/test_mappings.py b/backend/tests/unit/message_bus/test_mappings.py
index ffae6c3ce..83675c692 100644
--- a/backend/tests/unit/message_bus/test_mappings.py
+++ b/backend/tests/unit/message_bus/test_mappings.py
@@ -1,3 +1,10 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import pytest
+from prefect import Flow
+
 from infrahub.message_bus.messages import MESSAGE_MAP
 from infrahub.message_bus.operations import COMMAND_MAP

@@ -7,3 +14,14 @@ def test_message_command_overlap():
     messages = sorted(list(MESSAGE_MAP.keys()))
     commands = sorted(list(COMMAND_MAP.keys()))
     assert messages == commands
+
+
+@pytest.mark.parametrize(
+    "operation",
+    [pytest.param(function, id=key) for key, function in COMMAND_MAP.items()],
+)
+def test_operations_decorated(operation: Callable):
+    if callable(operation) and hasattr(operation, "__name__") and "Flow" not in type(operation).__name__:
+        pytest.fail(f"{operation.__name__} is not decorated with @flow")
+    else:
+        assert isinstance(operation, Flow), f"{operation.__name__} is not a valid Prefect flow"
```

dgarros committed:
Add flow decorator to all tasks

dgarros commented:
> What do you think of adding something like this? Looks like some tasks doesn't have the decorator but I don't know if this is on purpose.

Great idea, I added your test and I added the @flow on all the missing tasks
Initially I wanted to skip some of them but ultimately it's best to have some tests in place

dgarros committed:
Remove flow for refresh.registry.* tasks

dgarros committed:
Fix format
